### ISSUE TITLE
Disallow duplicate entries in packages

### ISSUE THREAD

**Issue by DemiMarie** [rpm-software-management/rpm#1905]
If I understand RPM’s source code correctly, it is possible for the same header tag to appear more than once in a package.  This makes little sense and should not be allowed.  I propose that RPM reject packages where the same tag appears more than once in the same header.

One way to achieve this is to do the following when loading a package from a file, as opposed to a header from the RPMDB:

- For each entry in the dribble, call findEntry() on the immutable region, and return an error if it finds something.
- After sorting the entries in the immutable region, check if there are two consecutive entries with the same tag.  If so, error out.

**Comment by pmatilai** [comment]
Yeah it's always been permitted, and it does seem bizarre, maybe there once was some use-cases depending on the behavior.
